proposal for robustness for I2C setSpeed()

Triggered today by this warning

```cpp
...Adafruit_BusIO\Adafruit_I2CDevice.cpp: In member function 'bool Adafruit_I2CDevice::setSpeed(uint32_t)':
...Adafruit_BusIO\Adafruit_I2CDevice.cpp:267:11: warning: variable 'prescaler' set but not used [-Wunused-but-set-variable]
  uint8_t prescaler = 1; 
```

Code from Adafruit_I2CDevice.cpp  263++   (some comments are mine)
```cpp
bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
#if defined(__AVR_ATmega328__) ||                                              \
    defined(__AVR_ATmega328P__) // fix arduino core set clock
  // calculate TWBR correctly

  uint8_t prescaler = 1;     //   <<<<<<<<<<<< 1
  uint32_t atwbr = ((F_CPU / desiredclk) - 16) / 2;     //  <<<<<<<< 3
  if (atwbr <= 255) {
    prescaler = 1;
    TWSR = 0x0;
  } else if (atwbr <= 1020) {
    atwbr /= 4;
    prescaler = 4;
    TWSR = 0x1;
  } else if (atwbr <= 4080) {
    atwbr /= 16;
    prescaler = 16;
    TWSR = 0x2;
  } else if (atwbr <= 16320) {
    atwbr /= 64;
    prescaler = 64;
    TWSR = 0x3;
  }
 
  //  what if atwbr > 16320  ??   //  <<<<<<<<<<<< 2

#ifdef DEBUG_SERIAL
  Serial.print(F("TWSR prescaler = "));
  Serial.println(prescaler);
  Serial.print(F("TWBR = "));
  Serial.println(atwbr);
#endif
  TWBR = atwb
  return true;
```


There are some problems, 
1. a warning, that **prescaler** is not used. (MINOR)
2. an error when **atwbr > 16320**, there is no error message.  (CLOCK UNDERFLOW)
3. an error when **(F_CPU / desiredclk < 18**   => TWBR could become 0 or "negative"  (CLOCK OVERFLOW)

Possible rewrite to tackle the above.
```cpp
bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
#if defined(__AVR_ATmega328__) ||                                              \
    defined(__AVR_ATmega328P__) // fix arduino core set clock
  // calculate TWBR correctly

  if ((F_CPU / 18) < desiredclk) {
    //  Serial.println(F("I2C.setSpeed too high."));     //  or some error = I2C_SETSPEED_OVERFLOW;
    return false;
  }
  uint32_t atwbr = ((F_CPU / desiredclk) - 16) / 2;
  if (atwbr > 16320) {
    //  Serial.println(F("I2C.setSpeed too low."));    //  or some error = I2C_SETSPEED_UNDERFLOW;
    return false;
  }
  
  if (atwbr <= 255) {
    atwbr /= 1;
    TWSR = 0x0;
  } else if (atwbr <= 1020) {
    atwbr /= 4;
    TWSR = 0x1;
  } else if (atwbr <= 4080) {
    atwbr /= 16;
    TWSR = 0x2;
  } else {           //  if (atwbr <= 16320)
    atwbr /= 64;
    TWSR = 0x3;
  }
  TWBR = atwb;

#ifdef DEBUG_SERIAL
  Serial.print(F("TWSR prescaler = "));
  Serial.println(pow(4, TWSR));
  Serial.print(F("TWBR = "));
  Serial.println(atwbr);
#endif
  return true;
```

--- a/Adafruit_I2CDevice.cpp
+++ b/Adafruit_I2CDevice.cpp
@@ -264,33 +264,43 @@ bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
 #if defined(__AVR_ATmega328__) ||                                              \
     defined(__AVR_ATmega328P__) // fix arduino core set clock
   // calculate TWBR correctly
-  uint8_t prescaler = 1;
+
+  if ((F_CPU / 18) < desiredclk) {
+    #ifdef DEBUG_SERIAL
+    Serial.println(F("I2C.setSpeed too high."));
+    #endif
+    return false;
+  }
   uint32_t atwbr = ((F_CPU / desiredclk) - 16) / 2;
+  if (atwbr > 16320) {
+    #ifdef DEBUG_SERIAL
+    Serial.println(F("I2C.setSpeed too low."));
+    #endif
+    return false;
+  }
+  
   if (atwbr <= 255) {
-    prescaler = 1;
+    atwbr /= 1;
     TWSR = 0x0;
   } else if (atwbr <= 1020) {
     atwbr /= 4;
-    prescaler = 4;
     TWSR = 0x1;
   } else if (atwbr <= 4080) {
     atwbr /= 16;
-    prescaler = 16;
     TWSR = 0x2;
-  } else if (atwbr <= 16320) {
+  } else {           //  if (atwbr <= 16320)
     atwbr /= 64;
-    prescaler = 64;
     TWSR = 0x3;
   }
+  TWBR = atwb;
+
 #ifdef DEBUG_SERIAL
   Serial.print(F("TWSR prescaler = "));
-  Serial.println(prescaler);
+  Serial.println(pow(4, TWSR));
   Serial.print(F("TWBR = "));
   Serial.println(atwbr);
 #endif
-  TWBR = atwbr;
   return true;
-
 #elif (ARDUINO >= 157) && !defined(ARDUINO_STM32_FEATHER) &&                   \
     !defined(TinyWireM_h)
   _wire->setClock(desiredclk);

--- a/Adafruit_I2CDevice.cpp
+++ b/Adafruit_I2CDevice.cpp
@@ -266,19 +266,19 @@ bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
   // calculate TWBR correctly
 
   if ((F_CPU / 18) < desiredclk) {
-    #ifdef DEBUG_SERIAL
+#ifdef DEBUG_SERIAL
     Serial.println(F("I2C.setSpeed too high."));
-    #endif
+#endif
     return false;
   }
   uint32_t atwbr = ((F_CPU / desiredclk) - 16) / 2;
   if (atwbr > 16320) {
-    #ifdef DEBUG_SERIAL
+#ifdef DEBUG_SERIAL
     Serial.println(F("I2C.setSpeed too low."));
-    #endif
+#endif
     return false;
   }
-  
+
   if (atwbr <= 255) {
     atwbr /= 1;
     TWSR = 0x0;
@@ -288,7 +288,7 @@ bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
   } else if (atwbr <= 4080) {
     atwbr /= 16;
     TWSR = 0x2;
-  } else {           //  if (atwbr <= 16320)
+  } else { //  if (atwbr <= 16320)
     atwbr /= 64;
     TWSR = 0x3;
   }

--- a/Adafruit_I2CDevice.cpp
+++ b/Adafruit_I2CDevice.cpp
@@ -292,7 +292,7 @@ bool Adafruit_I2CDevice::setSpeed(uint32_t desiredclk) {
     atwbr /= 64;
     TWSR = 0x3;
   }
-  TWBR = atwb;
+  TWBR = atwbr;
 
 #ifdef DEBUG_SERIAL
   Serial.print(F("TWSR prescaler = "));

